### PR TITLE
[ADD] stock_barcodes_gs1: Add support to stock_barcodes module to read GS1 barcodes

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please check Travis

**Comment by pedrobaeza** [comment]
@ernestotejeda @victormmtorres please review this

**Comment by sergio-teruel** [comment]
Mhuuu... it seems that DB connection is failing
https://travis-ci.org/OCA/stock-logistics-barcode/jobs/577395248#L364

**Comment by sergio-teruel** [comment]
Travis template updated

**Comment by sergio-teruel** [comment]
Fails due to product_multi_ean module that only alow to have EAN13 primary barcodes...

**Comment by pedrobaeza** [comment]
That's not valid anymore when field was generalized. That module must be changed.

**Comment by sergio-teruel** [comment]
This PR https://github.com/OCA/stock-logistics-barcode/pull/205 is needed for pass the tests

**Review by rafaelbn — APPROVED** [review]
:+1: tested functional in test enviroment

**Comment by rafaelbn** [comment]
Please @carlosdauden review thanks

File: stock_barcodes_gs1/wizard/stock_production_lot.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        except Exception:
+            return super().on_barcode_scanned(barcode)
+        package_barcode = barcode_decoded.get('01', False)
+        product_barcode = barcode_decoded.get('02', False)
+        lot_barcode = barcode_decoded.get('10', False)
+        if not lot_barcode:
+            return
+        if package_barcode:
+            packaging = self.env['product.packaging'].search([
+                ('barcode', '=', package_barcode),
```
Comment (rousseldenis) [inline stock_barcodes_gs1/wizard/stock_production_lot.py:25]: Shouldn't be useful to reuse _barcode_domain() function as in https://github.com/OCA/stock-logistics-barcode/pull/203/files#diff-fb62e484a3862c535a01cefb6a9d013cR45 ?

File: stock_barcodes_gs1/wizard/stock_production_lot.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            ])
+            if packaging:
+                self.product_id = packaging.product_id
+        elif product_barcode:
+            product = self.env['product.product'].search([
+                ('barcode', '=', product_barcode),
```
Comment (rousseldenis) [inline stock_barcodes_gs1/wizard/stock_production_lot.py:31]: Same as above.

**Review by rousseldenis — APPROVED** [review]
Code review. LGTM. Just a question

**Comment by sergio-teruel** [comment]
Hi @rousseldenis !!! Changes done... I can't be there this year !!!

**Review by carlosdauden — APPROVED** [review]

**Comment by sergio-teruel** [comment]
@rousseldenis the method _prepare_barcode_domain is not defined in wiz.stock.barcodes.new.lot so I can not use it in this model, but You are rigth, so in other PR, after this has been merged, I will do this improvement ... OK?

**Pull request merged**